python/ray/serve/controller.py

```diff
@@ -425,6 +426,17 @@ class Checkpoint:
     # TODO(ilr) Rename reconciler to PendingState
 
 
+@dataclass
+class FutureResult:
+    # Whether API call is successful
+    success: bool
+    # Goal requested when this future was created
+    requested_goal: Dict[str, Any]
+    # Goal state that supercedes (cancels) the requested goal
+    successor_goal: Optional[Dict[str, Any]] = field(
```

Comment:
I don't think we plan to expose this in the API as part of the MVP implementation so I'd prefer to remove this to avoid unnecessary complexity

Reply:
Actually, what will the requested_goal be used for? I'm not sure that we need that either.

Reply:
I can drop `requested_goal` for the moment. It would be used when we don't have the `create_event_and_set_result` logic, but we may keep that state outside of this object!

Reply:
Actually, I was going to check it as well on the API side--as a sanity check that what is being returned is correct.